content/en/docs/features/monitoring-and-diagnostics/monitoring-device-connectivity.md

```diff
@@ -0,0 +1,89 @@
+---
+title: 'Monitoring Device Connectivity'
+description: 'Learn how to determine the online or offline status of connected devices'
+date: '2023-08-08'
+categories: [features,iot-connectivity]
+keywords: [service, coap-gateway, network-monitoring]
+weight: 2
+---
+
+The CoAP gateway is responsible for the connection management including the up to date representation of the device status. This information is driven through the Resource Aggregate command - `UpdateDeviceMetadata`, that gets initiated when device connection status changes - device connects or disconnects from the CoAP Gateway. However, there are various scenarios to consider, such as unexpected termination of the CoAP Gateway, due to factors like node failure, power loss or others.
+
+## Managing Service Status Updates
+
+In the context of managing CoAP gateways and their associated devices, the process of setting devices to an offline state involves a series of steps:
+
+1. **Generation of Unique IDs**: Each instance of a CoAP gateway initiates with a unique identification (ID) generated during its initialization. These IDs play a dual role – they are stored in a database for reference and are also associated with device metadata. This association allows for streamlined updates regarding the online status of the devices.
+
+2. **RecordID Calculation**: To determine the appropriate `recordID` for updating the record within a N-minute period, a Golang logic is utilized:
+
+   ```golang
+   now := time.Now().UnixNano()
+   period := N * time.Minute
+   recordID = now - (now % int64(period))
+   ```
+
+3. **Shared Record Updating**: Regularly, each CoAP gateway instance contributes to a shared record among CoAP gateways in the database by appending its unique ID. The record being updated changes cyclically over time. Both the current shared record and the previous one are retained in the database for historical reference. The structure of the record is defined by the `SharedRecord` type in Golang:
+
+   ```golang
+   type SharedRecord struct {
+       RecordID          int64    `bson:"_id"`            // ID of the record moving forward in time
+       CoAPGatewayIDs    []string `bson:"coapgatewayids"` // IDs of lived CoAP gateways during the period
```

Comment:
Is the array checked for duplicates or how do we prevent the array from balooning to a giant size?

Reply:
The record is drop after some time. It is used only for one iteration.